### PR TITLE
[tensor] customized op returns ColoTensor

### PR DESCRIPTION
**Pull request by feifeibear**
We have to make the output of Op a ColoTensor.
Therefore we can assign the spec for them.
For example, the output of 1Drow Linear should be a shard of the global tensor. We can better represent it using ColoTensor than a Torch Tensor.

### CODE REVIEW

**Comment by feifeibear** [comment]
image

File: tests/test_tensor/test_op.py [hunk 1]
```diff
@@ -53,11 +53,11 @@ def test_linear():
 
     # torch.nn.functional.linear(torch.randn(1, in_dim), sharded_weight, sharded_bias)
     out = fc(input_tensor)
-    loss = out.sum()
```
Comment (Wesley-Jzy) [inline tests/test_tensor/test_op.py:56]: How will we support torch tensor's functionality? Implement all of them in the ColoTensor?

Comment (feifeibear) [reply tests/test_tensor/test_op.py:56]: Maybe, implement all of them in ColoTensor.

File: colossalai/tensor/_ops/linear.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+from numpy import isin
```
Comment (Wesley-Jzy) [inline colossalai/tensor/_ops/linear.py:1]: Auto added by vscode?

Comment (feifeibear) [reply colossalai/tensor/_ops/linear.py]: rm it

**Review by Wesley-Jzy — APPROVED** [review]

**Pull request merged**